corehq/apps/enterprise/iterators.py

```diff
@@ -1,7 +1,197 @@
+from datetime import datetime, timedelta
+from django.utils.translation import gettext as _
+from corehq.apps.es import filters
+from corehq.apps.es.utils import es_format_datetime
+from corehq.apps.es.forms import FormES
+from corehq.apps.enterprise.exceptions import TooMuchRequestedDataError
+from corehq.apps.app_manager.dbaccessors import get_brief_apps_in_domain
+
+
 def raise_after_max_elements(it, max_elements, exception=None):
     for total_yielded, ele in enumerate(it):
         if total_yielded >= max_elements:
             exception = exception or Exception('Too Many Elements')
             raise exception
 
         yield ele
+
+
+class IterableEnterpriseFormQuery:
+    '''
+    A class representing a query that returns its results as an iterator
+    The intended use case is to support queries that cross pagination boundaries
+    '''
```

Comment:
expand this docstring to explain `last_domain`, `last_time` and `last_id`. maybe an example?